tests/src/application/controllers/change-password.spec.ts

```diff
@@ -0,0 +1,162 @@
+import { type ChangePassword } from '@/domain/use-cases'
+import { type PasswordValidator } from '@/application/validation'
+import { ChangePasswordController } from '@/application/controllers'
+import { MissingParamError, NotFoundError, PasswordMismatchError, PasswordRequirementsError } from '@/application/errors'
+import { badRequest, success } from '@/application/helpers'
+import { makeFakeChangePasswordRequest, makeFakeChangePasswordUseCase, makeFakeServerError, makePasswordValidator } from '@/tests/utils'
+
+interface SutTypes {
+  sut: ChangePasswordController
+  changePasswordStub: ChangePassword
+  passwordValidatorStub: PasswordValidator
+}
+
+const makeSut = (): SutTypes => {
+  const changePasswordStub = makeFakeChangePasswordUseCase()
+  const passwordValidatorStub = makePasswordValidator()
+  const dependencies: ChangePasswordController.Dependencies = {
+    changePassword: changePasswordStub,
+    passwordValidator: passwordValidatorStub
+  }
+  const sut = new ChangePasswordController(dependencies)
+  return {
+    sut,
+    changePasswordStub,
+    passwordValidatorStub
+  }
+}
+
+describe('ChangePasswordController', () => {
+  describe('test the id field', () => {
+    it('Should return 400 if no userId is provided', async () => {
+      const { sut } = makeSut()
+
+      const httpRequest = makeFakeChangePasswordRequest({ withUserId: false })
+      const httpResponse = await sut.handle(httpRequest)
+
+      expect(httpResponse).toEqual(badRequest(new MissingParamError('id')))
+    })
+  })
+
+  describe('test the password field', () => {
+    it('Should return 400 if no password is provided', async () => {
+      const { sut } = makeSut()
+
+      const httpRequest = makeFakeChangePasswordRequest({ fields: { password: undefined } })
+      const httpResponse = await sut.handle(httpRequest)
+
+      expect(httpResponse).toEqual(badRequest(new MissingParamError('password')))
+    })
+
+    it('Should return 400 if an invalid password is provided', async () => {
+      const { sut, passwordValidatorStub } = makeSut()
+      jest.spyOn(passwordValidatorStub, 'isValid').mockReturnValueOnce(false)
+
+      const httpRequest = makeFakeChangePasswordRequest({
+        fields: {
+          password: 'invalid_password',
+          passwordConfirmation: 'invalid_password'
+        }
+      })
+      const httpResponse = await sut.handle(httpRequest)
+
+      expect(httpResponse).toEqual(badRequest(new PasswordRequirementsError()))
+    })
+
+    it('Should call PasswordValidator with correct password', async () => {
+      const { sut, passwordValidatorStub } = makeSut()
+      const isValidSpy = jest.spyOn(passwordValidatorStub, 'isValid')
+
+      const httpRequest = makeFakeChangePasswordRequest({
```

Comment:
Why this `makeFakeChangePasswordRequest`? You Just omit 'body' key. 

If you want use the semantic of 'valid_password' or 'invalid_password' does not make sense use this `makeFake` objects. Because I understand this objects how a way to reduce unnecessary lines, this is a trade of between lines of code and be explicitly. Make sense?